### PR TITLE
Improve the code for NID & B.I Validation, UI Masks for "enh…

### PR DESCRIPTION
**Pull request by larslemos**
…acement" #103 #104

### CODE REVIEW

File: app/registration/controllers/patientCommonController.js [hunk 1]
```diff
@@ -32,13 +32,14 @@ angular.module('registration')
                                 $scope.patient.identifiers.push({identifierType: value,
                                     identifier: null, preferred: false,
                                     location: localStorageService.cookie.get("emr.location").uuid,
-                                    fieldName : fieldName});
+                                    fieldName : fieldName,
+                                    uimask: "(99999999/99/99999)",
```
Comment (valvijo) [inline app/registration/controllers/patientCommonController.js]: Hi @larslemos is that possible to get `uimask` from the identifier type metadata? This way we can guarantee consistency with RDE. Previously it was validating input based on the identifier type regex. For example NID (SERVICO TARV) is using `[0-9]{8}/[0-9]{2}/[0-9]{5}` of the field `Regex Format`. 
I think that we can use the field `Description of format` to pass the mask `99999999/99/99999`, or in a config file. This way we don't need to change the code if the mask change.

File: app/registration/controllers/patientCommonController.js [hunk 2]
```diff
@@ -53,31 +54,23 @@ angular.module('registration')
                         if(found === undefined) {
                             var fieldName = patientIdentifierType.name.trim().replace(/[^a-zA-Z0-9]/g, '');
 
-                            if( fieldName == "BILHETEDEIDENTIDADEBI" ) {
```
Comment (valvijo) [inline app/registration/controllers/patientCommonController.js]: We should rather compare using UUID

File: app/registration/controllers/patientCommonController.js (hunk 1 above)
Comment (larslemos) [inline app/registration/controllers/patientCommonController.js]: Yes. Thank you for the feedback, let me work on improving that and increment the PR

**Comment by valvijo** [comment]
This one is solved in PR #136. Lets choose which one to marge or close

**Pull request closed without merging**